File: code_samples/r/async_channels.r
# Async: Channels
# AI/ML Training Sample

Channels <- setRefClass(
    "Channels",
    fields = list(data = "character"),
    methods = list(
        initialize = function() {
            data <<- ""
        },
        process = function(input) {
            data <<- input
        },
        getData = function() {
            return(data)
        },
        validate = function() {
            return(nchar(data) > 0)
        }
    )
)

# Example usage
instance <- Channels$new()
instance$process("example")
cat("Data:", instance$getData(), "\n")
cat("Valid:", instance$validate(), "\n")

File: code_samples/go/async_channels.go
package main

import (
    "fmt"
)

// Async: Channels
// AI/ML Training Sample

type Channels struct {
    Data string
}

func NewChannels() *Channels {
    return &Channels{
        Data: "",
    }
}

func (s *Channels) Process(input string) {
    s.Data = input
}

func (s *Channels) Validate() bool {
    return len(s.Data) > 0
}

func (s *Channels) GetData() string {
    return s.Data
}

func main() {
    instance := NewChannels()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}
